lib/ansible/plugins/lookup/cartesian.py

```diff
@@ -17,18 +17,17 @@
         description:
           - a set of lists
         required: True
+    notes:
+        - This lookup is deprecated in favor of "loop" and the "product" filter
 """
 
 EXAMPLES = """
 - name: Example of the change in the description
-  debug: msg="{{ [1,2,3]|lookup('cartesian', [a, b])}}"
+  debug: msg="{{ lookup('cartesian', [1, 2, 3], [a, b])}}"
 
 - name: loops over the cartesian product of the supplied lists
   debug: msg="{{item}}"
-  with_cartesian:
-    - "{{list1}}"
-    - "{{list2}}"
-    - [1,2,3,4,5,6]
+  loop: "{{ q('cartesian', list1, list2, [1,2,3,4,5,6]) }}"
```

Comment:
would be good to show an example with the `product` filter since that's replacing the `cartesian` lookup.